assets/src/BlockEditorValidation.js

```diff
@@ -0,0 +1,96 @@
+const { registerPlugin } = wp.plugins;
+const { PluginPrePublishPanel } = wp.editPost;
+const { dispatch } = wp.data;
+const {__} = wp.i18n;
+
+export const blockEditorValidation = () => {
+  registerPlugin( 'pre-publish-checklist', { render: PrePublishCheckList } );
+};
+
+const PrePublishCheckList = () => {
+  let lockPost = false;
+
+  let checkListMsg = '';
+  let classname    = '';
+
+  // "Analytics & Tracking" fields validation.
+  let $htmlbody   = $( 'html, body' );
+  let $toValidate = $( '[data-validation]' );
+
+  let analyticsErrorFlag   = false;
+
+  if ( $toValidate.length ) {
+    let $row = null;
+
+    function add_required($row) {
```

Comment:
Can you move these function declarations outside of `PrePublishCheckList`? That will make it easier to read `PrePublishCheckList`.